File: 1111-maximum-nesting-depth-of-two-valid-parentheses-strings/1111-maximum-nesting-depth-of-two-valid-parentheses-strings.cpp
class Solution {
public:
    vector<int> maxDepthAfterSplit(string seq) {
        vector<int> ans;
        int depth = 0;
        for (auto ch : seq) {
            if (ch == '(') {
                ans.push_back(depth % 2);
                depth++;
            } else if (ch == ')') {
                depth--;
                ans.push_back(depth % 2);
            }
        }
        return ans;
    }
};

File: 1111-maximum-nesting-depth-of-two-valid-parentheses-strings/1111-maximum-nesting-depth-of-two-valid-parentheses-strings.java
class Solution {
    public int[] maxDepthAfterSplit(String seq) {
        int n = seq.length(), depth = 0;
        int[] ans = new int[n];
        int i=0;
        for(char c:seq.toCharArray()){
            if(c=='('){
                ans[i]=depth%2;
                depth++;
            }else{
                depth--;
                ans[i]=depth%2;
            }
            i++;
        }
        return ans;
    }
}